runtime/src/bank.rs

```diff
@@ -4451,10 +4451,117 @@ impl Bank {
         }
     }
 
-    fn get_account_overrides_for_simulation(&self, account_keys: &AccountKeys) -> AccountOverrides {
+    /// Run multiple sequential transactions against a frozen bank without committing the results
+    pub fn simulate_multiple_transactions(
+        &self,
+        transactions: Vec<SanitizedTransaction>,
+    ) -> Vec<TransactionSimulationResult> {
+        assert!(self.is_frozen(), "simulation bank must be frozen");
+        self.simulate_multiple_transactions_unchecked(transactions)
+    }
+
+    /// Run multiple sequential transactions against a bank without committing the results; does not check if the bank
+    /// is frozen
+    pub fn simulate_multiple_transactions_unchecked(
+        &self,
+        transactions: Vec<SanitizedTransaction>,
+    ) -> Vec<TransactionSimulationResult> {
+        let account_keys: Vec<_> = transactions
+            .iter()
+            .map(|tx| tx.message().account_keys())
+            .collect();
+        let account_nums = account_keys
+            .iter()
+            .map(|keys| keys.len())
+            .collect::<Vec<_>>();
+        let mut account_overrides = self.get_account_overrides_for_simulation(&account_keys);
+
+        let mut results = vec![];
+        for (transaction, number_of_accounts) in transactions.iter().zip(account_nums) {
+            let batch = self.prepare_unlocked_batch_from_single_tx(&transaction);
```

Comment:
optimization here could be locking as many non-conflicting transactions that you can + doing some batching